include/net/kfi/kfi_provider.h

```diff
@@ -33,18 +33,10 @@
 #ifndef _KFI_PROVIDER_H_
 #define _KFI_PROVIDER_H_
 
-#include <linux/kernel.h>
-#include <linux/byteorder/generic.h>
-#include <linux/semaphore.h>
-#include <linux/list.h>
-#include <linux/string.h>
```

Comment:
That is what I thought too: include as close to use as possible. This patch does a lot of that but a following can clean this up even more.